homeassistant/components/zoneminder.py

```diff
@@ -125,3 +136,30 @@ def get_state(api_url):
 def change_state(api_url, post_data):
     """Update a state using the Zoneminder API."""
     return _zm_request('post', api_url, data=post_data)
+
+
+# pylint: disable=no-member
+def get_active_state():
+    """Get the current (string) run state from Zoneminder API."""
+    active_state = None
+    for i in get_state('api/states.json')['states']:
+        # yes, the ZM API uses the *string* "1" for this...
+        if i['State']['IsActive'] == '1':
+            active_state = i['State']['Name']
+    return active_state
+
+
+def set_active_state(call):
+    """
+    Set the ZoneMinder run state to the given state name, via ZM API.
+
+    Note that this is a long-running API call; ZoneMinder changes the state of
+    each camera in turn, and this GET does not receive a response until all
+    cameras have been updated. Even on a reasonably powerful machine, this call
+    can take ten (10) or more seconds **per camera**. This method sets a
+    timeout of 120, which should be adequate for most users.
+    """
+    state_name = call.data.get('name')
+    url = 'api/states/change/%s.json' % state_name
+    _LOGGER.debug('Setting ZoneMinder run state via GET %s', url)
+    return _zm_request('GET', url, timeout=120)
```

Comment:
This looks like something that should be handled in a separate library. I'm not sure why we haven't asked for that earlier, but I think the interaction with the ZM api should be broken out into standalone library published on pypi.

Reply:
Possibly blocking a thread in our thread pool for 2 minutes is most likely not what we want.

Using aiohttp instead of requests would hopefully work and solve that problem.

This should be solved by the standalone library.

Reply:
Is there an existing process for breaking things out into standalone libraries?

I'm trying to contribute some improvements to the ZM integration which seem pretty central to how ZM works, but I really don't have any desire to maintain a separate Python library that wraps the ZM API, nor to implement enough of the API to make said project useful...

I guess I'll leave that up to you and other HASS folks, but if maintaining a ZM API client on PyPI is what it takes to get this merged, I'll just close the PR and continue running off of my fork... I'm actively working to _reduce_ the number of projects I maintain, and can't in good conscience take on another.